src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final double side;
+    private final ColorSupplier color;
+
+    public Square(ColorSupplier color, double side) {
```

Comment:
Don't pass expressions in the constructor. Create separate variables and pass them on for better code readability.